src/wayland/compositor/mod.rs

```diff
@@ -453,6 +458,21 @@ pub struct CompositorClientState {
     queue: Mutex<Option<TransactionQueue>>,
 }
 
+impl CompositorClientState {
+    pub fn blocker_cleared<D: CompositorHandler>(
+        &self,
+        surface: &WlSurface,
+        state: &mut D,
+        dh: &DisplayHandle,
+    ) {
+        if let Some(queue) = self.queue.lock().unwrap().as_mut() {
+            if queue.apply_ready(dh) {
+                CompositorHandler::commit(state, surface)
```

Comment:
I think we probably should update the documentation of `CompositorHandle::commit()` to reflect more precisely when I may be invoked.

Reply:
done